include/net/ethernet_mgmt.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Ethernet Management interface public header
+ */
+
+#ifndef __ETHERNET_MGMT_H__
+#define __ETHERNET_MGMT_H_
+
+#include <net/ethernet.h>
+#include <net/net_mgmt.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Ethernet library
+ * @defgroup ethernet_mgmt Ethernet Library
+ * @ingroup networking
+ * @{
+ */
+
+#define _NET_ETHERNET_LAYER	NET_MGMT_LAYER_L2
+#define _NET_ETHERNET_CODE	0x802
+#define _NET_ETHERNET_BASE	(NET_MGMT_IFACE_BIT |			\
+				 NET_MGMT_LAYER(_NET_ETHERNET_LAYER) |	\
+				 NET_MGMT_LAYER_CODE(_NET_ETHERNET_CODE))
+#define _NET_ETHERNET_EVENT	(_NET_ETHERNET_BASE | NET_MGMT_EVENT_BIT)
+
+enum net_request_ethernet_cmd {
+	NET_REQUEST_ETHERNET_CMD_SET_AUTO_NEGOCIATION = 1,
+	NET_REQUEST_ETHERNET_CMD_SET_LINK,
+	NET_REQUEST_ETHERNET_CMD_SET_DUPLEX,
+	NET_REQUEST_ETHERNET_CMD_SET_MAC_ADDRESS,
+};
+
+#define NET_REQUEST_ETHERNET_SET_AUTO_NEGOCIATION			\
+	(_NET_ETHERNET_BASE | NET_REQUEST_ETHERNET_CMD_SET_AUTO_NEGOCIATION)
+
+NET_MGMT_DEFINE_REQUEST_HANDLER(NET_REQUEST_ETHERNET_SET_AUTO_NEGOCIATION);
+
+#define NET_REQUEST_ETHERNET_SET_LINK					\
+	(_NET_ETHERNET_BASE | NET_REQUEST_ETHERNET_CMD_SET_LINK )
+
+NET_MGMT_DEFINE_REQUEST_HANDLER(NET_REQUEST_ETHERNET_SET_LINK);
+
+#define NET_REQUEST_ETHERNET_SET_DUPLEX					\
+	(_NET_ETHERNET_BASE | NET_REQUEST_ETHERNET_CMD_SET_DUPLEX)
+
+NET_MGMT_DEFINE_REQUEST_HANDLER(NET_REQUEST_ETHERNET_SET_DUPLEX);
+
+#define NET_REQUEST_ETHERNET_SET_MAC_ADDRESS				\
+	(_NET_ETHERNET_BASE | NET_REQUEST_ETHERNET_CMD_SET_MAC_ADDRESS)
+
+NET_MGMT_DEFINE_REQUEST_HANDLER(NET_REQUEST_ETHERNET_SET_MAC_ADDRESS);
+
+struct net_eth_addr;
+
+struct ethernet_req_params {
```

Comment:
This seems to be same as "struct ethernet_config". Wwhy not have just one  instead?

Reply:
It just happens to be the same right now, but if we add ethernet L2 specific requests/parameters, it will diverge.